R/pension_supplement.R

```diff
@@ -0,0 +1,103 @@
+#' Pension Supplement
+#' 
+#' The Pension Supplement gets added to the max rate of payment before income reduction tests are applied. Note that if the individual is part of a couple, the rate indicates the payment amount per person, not for the couple.
+#' Can be claimed by those receiving Age Pension, Carer Payment, Wife Pension, Widow B Pension, Bereavement Allowance, or Disability Support Pension (except if under 21 and have no children).
+#' Can also be claimed if over age pension age and are receiving ABSTUDY, Austudy, Parenting Payment, Partner Allowance, Special Benefit, or Widow Allowance.
+#' Can still claim the basic amount if single, under age pension age, and receive the Parenting Payment.
+#'
+#' @param has_partner Does the individual have a partner?
+#' @param age The individual's age. Default is 18 years.
+#' @param n_dependants How many dependant children does the individual have?
+#' @param parenting_payment Is the individual receiving parenting payment?
+#' @param Date Date. Default is "2016/03/01" if fy.year is not present.
+#' @param fy.year Financial year. Default is "2015-16" if Date is not present.
+#' @param age_pension_age_requirement Does the payment for which the pension supplement is being added to require the individual to be over age pension age to receive? i.e. one of ABSTUDY, Austudy, Parenting Payment, Partner Allowance, Special Benefit, Widow Allowance. 
```

Comment:
Don't understand the documentation of `age_pension_age_requirement`.

Reply:
Edited to: 'Is the individual required to be over age pension age to receive the supplement? i.e. one of ABSTUDY, Austudy, Parenting Payment, Partner Allowance, Special Benefit, Widow Allowance. '

Reply:
My reaction is to simply have an argument `age` and handle the logic in the body of the function. Is that possible, or is it independent of age? 

If it's independent then I think it would be best to still have age but replace this argument with an argument `qualifying_payment` which takes a string like `"austudy"` and handle the logic in connection with age and `age_pension_age`. That is if `qualifying_payment %in% c("Austudy", ...)`. If I understand correctly, the payment types that come with a supplement are unlikely to change.

Reply:
So to be eligible for the supplement you do not have to be `age_pension_age` for certain payments such as Wife Pension, but you do for others like Austudy. 

I'll add in the argument `qualifying_payment` now.